### PR TITLE
20221022 improve tests

### PR DESCRIPTION
**Pull request by Firstyear**
This is a huge cleanup focused on two areas. First is removal of the legacy test macros in favour of proc macos that init tests such that we also have an async runtime available  - this is required so we can remove async_std from the code base. The second is correcting a long standing defect with mutability in write transactions. We have a number of unsafe internal accesses because our transactions were not mutable wrt to rust, and so we need to use unsafe cell internally. This is a long standing hold over from older versions where async and other rust features were not possible or available. 

- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Review by yaleman — APPROVED** [review]
Curious about one thing, but otherwise... woot :D

File: kanidmd/core/src/actors/v1_write.rs [hunk 1]
```diff
@@ -346,7 +346,7 @@ impl QueryServerWriteV1 {
         filter: Filter<FilterInvalid>,
         eventid: Uuid,
     ) -> Result<(), OperationError> {
-        let idms_prox_write = self.idms.proxy_write_async(duration_from_epoch_now()).await;
+        let mut idms_prox_write = self.idms.proxy_write(duration_from_epoch_now()).await;
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply kanidmd/core/src/actors/v1_write.rs:349]: When an operation (event) starts, we need the "current time" to be identical throughout that event. So in some cases we have the current time only needed being to create the transaction itself (such as the above) and no other elements of the event take time. When current time is needed in multiple calls within the fn, we set ct to a value and re-use it throughout the operation.

A possible improvement here would actually be that when we start the txn we set ct, but then inside of the server, we access ct via the txn instead of from function args. That would guarantee that it's the "current time of the transaction" rather than relying on this "pattern" that currently exists.

So yeah, tl;dr - in any "event" we assume time is a fixed moment (instant) rather than calling to a clock source

The reason generally is that this lets us control exactly what "time" it is during tests so we can advance and rewind time as needed to test edge cases.

Hope that helps :)

[the remark this answers is not in the document]
Comment (yaleman) [reply kanidmd/core/src/actors/v1_write.rs:349]: I think I'm *more* confused now, but if it's for good reason, then so be it 😃
